unity/Holo/Assets/Editor/ModelImport/VTKImport/UnstructuredGridImporter.cs

```diff
@@ -0,0 +1,234 @@
+                if (verticesFlag & indicesFlag)
+                {
+                    break;
+                }
+
+                if (line.IndexOf("POINTS", StringComparison.CurrentCultureIgnoreCase) >= 0)
+                {
+                    string[] pointsData = line.Split(' ');
+                    int numberOfVertices = int.Parse(pointsData[1]);
+                    GetVertices(streamReader, numberOfVertices);
+                    verticesFlag = true;
+                }
+
+                if (line.IndexOf("CELLS", StringComparison.CurrentCultureIgnoreCase) >= 0 & verticesFlag == true)
+                {
+                    string[] cellsData = line.Split(' ');
+                    int numberOfLines = int.Parse(cellsData[1]);
+                    GetBodyIndicesAndNormals(streamReader, numberOfLines);
+                    indicesFlag = true;
+                }
+            }
+        }
+
+        //Get meshes' verices. 
+        private void GetVertices(StreamReader streamReader, int numberOfVertices)
+        {
+            Vertices = new Vector3[numberOfVertices];
+            bool firstVertex = true;
+            for (int i = 0; i < numberOfVertices; i++)
+            {
+                Vector3 currentVertex = streamReader.GetLineVertex();
+                Vertices[i] = currentVertex;
+                BoundingVertices.UpdateBoundingVertices(firstVertex, currentVertex);
+                firstVertex = false;
+            }
+        }
+
+        //Gets body indices in the file and calculates its normals.
+        private void GetBodyIndicesAndNormals(StreamReader streamReader, int numberOfLines)
+        {
+            List<int> indices = new List<int>();
+            List<int> facetIndices = new List<int>();
+
+            bool firstVertex = true;
+            for (int i = 0; i < numberOfLines; i++)
+            {
+                facetIndices = streamReader.GetLineIndices();
+                if (firstVertex)
+                {
+                    IndicesInFacet = facetIndices.Count;
+                }
+                if (facetIndices.Count > 2)
+                {
+                    UpdateFacetNormals(facetIndices);
+                }
+                indices.AddRange(facetIndices);
+            }
+            if (facetIndices.Count > 2)
+            {
+                foreach (Vector3 normal in Normals)
+                {
+                    normal.Normalize();
+                }
+            }
+            Indices = indices.ToArray();
+        }
+
+        //Updates facet normals.
+        private void UpdateFacetNormals(List<int> facetIndices)
+        {
+            Vector3 currentNormal = new Vector3();
+            currentNormal = CalculateFacetNormal(facetIndices);
+            foreach (int index in facetIndices)
+            {
+                Normals[index] += currentNormal;
+            }
+        }
+
+        //Calculates a normal of a facet.
+        private Vector3 CalculateFacetNormal(List<int> facetIndices)

```

Comment:
`facetIndices` is supposed to have 3 elements. You can put constraint on that by specifying the type.

Reply:
Changed to Array type.